pkg/queue/sharedmain/main.go

```diff
@@ -243,52 +250,47 @@ func Main(opts ...Option) error {
 		concurrencyendpoint = queue.NewConcurrencyEndpoint(env.ConcurrencyStateEndpoint, env.ConcurrencyStateTokenPath)
 	}
 
+	var srvs []service
+	var drainer *pkghandler.Drainer
+	var drainerTLS *pkghandler.Drainer
+	var mainServer service
```

Comment:
Is there a specific reason why we declare all the servers upfront?

Reply:
Yes, see the next comment.
I will reconfirm if indeed all are needed.